modules/relayers/src/payment_adapter.rs

```diff
@@ -19,22 +19,25 @@
 
 use crate::{Config, RelayerRewards};
 
-use bp_messages::source_chain::{MessageDeliveryAndDispatchPayment, RelayersRewards};
-use frame_support::sp_runtime::SaturatedConversion;
+use bp_messages::source_chain::{
+	DeliveryConfirmationPayments, MessageDeliveryAndDispatchPayment, RelayersRewards,
+};
+use frame_support::{sp_runtime::SaturatedConversion, traits::Get};
 use sp_arithmetic::traits::{Saturating, UniqueSaturatedFrom, Zero};
 use sp_std::{collections::vec_deque::VecDeque, marker::PhantomData, ops::RangeInclusive};
 
 /// Adapter that allows relayers pallet to be used as a delivery+dispatch payment mechanism
 /// for the messages pallet.
-pub struct MessageDeliveryAndDispatchPaymentAdapter<T, MessagesInstance>(
-	PhantomData<(T, MessagesInstance)>,
+pub struct DeliveryConfirmationPaymentsAdapter<T, DeliveryReward, ConfirmationReward>(
+	PhantomData<(T, DeliveryReward, ConfirmationReward)>,
 );
 
-impl<T, MessagesInstance> MessageDeliveryAndDispatchPayment<T::RuntimeOrigin, T::AccountId>
-	for MessageDeliveryAndDispatchPaymentAdapter<T, MessagesInstance>
+// TODO (https://github.com/paritytech/parity-bridges-common/issues/1318): this impl must be removed
```

Comment:
#1318 is closed.. should we open another dedicated one?

Also sorry for dumb q, this comment is tracking the removal of `MessageDeliveryAndDispatchPayment` trait not just this impl, right? Since we don't have message dispatch anymore. Or will we still use this and rename to `trait MessageDeliveryPayment`?

Reply:
I'm gonna remove impl (`DeliveryConfirmationPaymentsAdapter`) and will keep the other impl (`DeliveryConfirmationPaymentsAdapter`). The old trait also needs to be removed

Reply:
Sounds good! It's clear now, thanks!